### PR TITLE
chore: add .golangci config to add common used checks to help us improve the quality

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

The golangci default config does not check all that is possible
This PR is adding common usage checks and sorting out some issues found with the new lints

PS.: See that some of them are commented/disabled so that we can sort those in a follow up to make easier the review.

### CODE REVIEW

File: .golangci.yml
Comment (camilamacedo86) [inline .golangci.yml:71]:
```suggestion
     - unused

```

**Review by ricardomaraschini — COMMENTED** [review]
You have my LGTM here for sure. Let's just wait for someone else's review as well.

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**
